### PR TITLE
Modernize

### PR DESCRIPTION
**Pull request by luis-pereira**
Use C++11 features.
Done with `clang-tidy`.

### CODE REVIEW

File: src/qtxdg/xdgdesktopfile.cpp [hunk 1]
```diff
@@ -986,7 +986,7 @@ static QStringList parseCombinedArgString(const QString &program)
 void replaceVar(QString &str, const QString &varName, const QString &after)
 {
     str.replace(QRegExp(QString::fromLatin1("\\$%1(?!\\w)").arg(varName)), after);
-    str.replace(QRegExp(QString::fromLatin1("\\$\\{%1\\}").arg(varName)), after);
+    str.replace(QRegExp(QString::fromLatin1(R"(\$\{%1\})").arg(varName)), after);
```
Comment (palinek) [inline src/qtxdg/xdgdesktopfile.cpp]: ...and why not in the line above? :)

**Comment by tsujan** [comment]
Nice!

(1) Raw string literals have problem with translation updates with Qt < 5.12.2 (I think @palinek fixed them in Qt). Do we require that version?
(2) Now that the code is updated, why not replacing `QRegExp` with `QRegularExpression` too?

**Comment by palinek** [comment]
> (1) Raw string literals have problem with translation updates with Qt < 5.12.2 (I think @palinek fixed them in Qt). Do we require that version?

It seems, it is backported into versions from 5.12 -> https://github.com/qt/qttools/commit/ee9f6ec9b9e8f0d18ad7e8fededb5292abb91912

**Comment by tsujan** [comment]
> It seems, it is backported into versions from 5.12

I recompiled it with your patch since Qt 5.11 (needed it for my apps) but, only when 5.12.2 was released, I didn't need a recompilation.

Anyhow, it'll be OK if there's nothing to translate in files with raw string literals.

**Comment by luis-pereira** [comment]
> Anyhow, it'll be OK if there's nothing to translate in files with raw string literals.

That's the case.

**Comment by tsujan** [comment]
> That's the case.

Good.

File: src/qtxdg/xdgdesktopfile.cpp (hunk 1 above)
Comment (luis-pereira) [reply src/qtxdg/xdgdesktopfile.cpp]: clang-tidy, probably, got confused by the parentheses.
I only checked what it changed.

``` cpp
R"regexp(\$%1(?!\w))regexp"
```
cuts it.

Will add ASAP.

**Pull request merged**